interfaces/devdoc/file_util_linux.md

```diff
@@ -0,0 +1,67 @@
+# file_util_linux
+
+## Overview
+
+Linux implementation of the file operation functions.
+
+## Exposed API 
+
+```c
+// Most defines removed for space constraints
+#define INVALID_HANDLE_VALUE            NULL
+
+typedef void *HANDLE;
+typedef void *LPSECURITY_ATTRIBUTES;
+
+MOCKABLE_FUNCTION(, HANDLE, ebs_create_file, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
```

Comment:
>ebs_create_file [](http://example.com/codeflow?start=28&length=15)

Can we have some black magic where the user would write in the bsdl code CreateFile? and we redirect the calls to whatever APIs needed.

That brings the benfit that there are no changes needed in the user layer and also one can actually hit F1 in Visual studio and get help on the APIs right away.

Reply:
Yes! We renamed the functions to ebs_create_file and same thing for close handle so it can redirect into Windows and call win32 API.